patches/com/mojang/blaze3d/opengl/GlDevice.java.patch

```
@@ -0,0 +1,68 @@
++            if (GlConst.toGlInternalId(candidate) == internalFormat || GlConst.toGlExternalId(candidate) == internalFormat) {
++                format = candidate;
++                break;
++            }
++        }
++
++        if (format == null) {
++            org.lwjgl.opengl.GL11.glBindTexture(org.lwjgl.opengl.GL11.GL_TEXTURE_2D, previousBinding);
++            throw new IllegalArgumentException("Couldn't find a matching vanilla TextureFormat for OpenGL internal format id " + internalFormat);
++        }
++
++        // Get width and height
++        int width = org.lwjgl.opengl.GL11.glGetTexLevelParameteri(org.lwjgl.opengl.GL11.GL_TEXTURE_2D, 0, org.lwjgl.opengl.GL11.GL_TEXTURE_WIDTH);
++        int height = org.lwjgl.opengl.GL11.glGetTexLevelParameteri(org.lwjgl.opengl.GL11.GL_TEXTURE_2D, 0, org.lwjgl.opengl.GL11.GL_TEXTURE_HEIGHT);
++
++        // Count mip levels by querying sizes until we get 0
++        int mipLevels = 1;
++        int mipWidth = width;
++        while (mipWidth > 1) {
++            int nextWidth = org.lwjgl.opengl.GL11.glGetTexLevelParameteri(org.lwjgl.opengl.GL11.GL_TEXTURE_2D, mipLevels, org.lwjgl.opengl.GL11.GL_TEXTURE_WIDTH);
++            if (nextWidth == 0) {
++                break;
++            }
++            mipLevels++;
++            mipWidth = nextWidth;
++        }
++
++        org.lwjgl.opengl.GL11.glBindTexture(org.lwjgl.opengl.GL11.GL_TEXTURE_2D, previousBinding);
++
++        return createExternalTexture(label, format, width, height, mipLevels, nativeId);
++    }
++
++    /**
++     * Create a new GPU texture from an existing texture whose lifecycle is externally managed.
++     */
++    public GpuTexture createExternalTexture(@Nullable String label, TextureFormat format, int width, int height, int mipLevels, int nativeId) {
++        if (label == null) {
++            label = String.valueOf(nativeId);
++        }
++        GlTexture gltexture = new GlTexture(label, format, width, height, mipLevels, nativeId);
```

the wrong GlTexture constructor is called here, texture id is deleted twice. note: this is not a violation of the GL spec. glDeleteTextures silently ignores non-texture ids, hence no error is generated.

---

Addressed